pandas/_libs/tslibs/period.pyx

```diff
@@ -1120,9 +1120,12 @@ cdef class _Period(object):
                 if other.freq != self.freq:
                     msg = _DIFFERENT_FREQ.format(self.freqstr, other.freqstr)
                     raise IncompatibleFrequency(msg)
-                return self.ordinal - other.ordinal
+                return (self.ordinal - other.ordinal) * self.freq
             elif getattr(other, '_typ', None) == 'periodindex':
-                return -other.__sub__(self)
+                # GH#21314 PeriodIndex - Period returns an object-index
+                # of DateOffset objects, for which we cannot use __neg__
+                # directly, so we have to apply it pointwise
```

Comment:
is there a test that hits this explicity?

Reply:
Yes.  Without this we get test failures in `tests.indexes.period.test_arithmetic.TestPeriodIndexSeriesMethods` for 
`test_pi_ops`, `test_pi_sub_period`, `test_pi_sub_period_nat`